### DISCUSSION TITLE
Is it possible to get native window handle from slint window?

### DISCUSSION THREAD

**Discussion by gibberishCode** [slint-ui/slint#5107] [Q&A]
I see that there is a way to connect native window to the slint https://github.com/slint-ui/slint/tree/master/examples/cpp/platform_native
but I just want to get native handler somehow and can't find a way to do so. Is it possible?

**Comment by tronical** [comment]
The git version has a way to obtain a raw window handle in Rust:

https://snapshots.slint.dev/master/docs/rust/slint/struct.window#method.window_handle

is that what you’re looking for?

**Reply by gibberishCode** [reply]
Yes this is it, but I am using c++, i found method with the same name in c++ but it leads to something private to slint and has nothing to do with os specific window handle, if i get it right

**Reply by ogoffart** [reply]
There is no C++ API for that at this point.
We probably should expose this though.

(We currently have slint::platform::NativeWindowHandle, bur it's only used to "set" the window handle to the renderer, not to get it. If we want to use NativeWindowHandle for that, we should move it out of the platform namespace)

**Reply by ogoffart** [reply]
If you use the Qt backend, you can access the QWidget*

**Comment by sollapse** [comment]
Is there any feedback on whether exposing the native window handle will be allowed in the C++ API?

**Reply by tronical** [reply]
A kind contributor implemented this recently. It’s in the master branch and will be part of next release. Check out the api at the bottom of this page: https://snapshots.slint.dev/master/docs/cpp/api/classslint_1_1window#exhale-class-classslint-1-1window
